app/javascript/shared/components/keyboardShortcuts.jsx

```diff
@@ -0,0 +1,66 @@
+import PropTypes from 'prop-types';
+import { h } from 'preact';
+import { useEffect } from 'preact/hooks';
+
+/**
+ * Component that can be added to the dom to listen 
+ * for keyboard presses
+ * 
+ * @example
+ * const shortcuts = {
+ *   "ctrl+alt+KeyG": (e) => {
+ *     e.preventDefault(); 
+ *     alert("Control Alt G has been pressed")
+ *   }
+ * }
+ * <KeyboardShortcuts shortcuts={shortcuts} />
+ * 
+ */
+const KeyboardShortcuts = ({ shortcuts, ...props }) => {
+  useEffect(() => {
+    // Return true if element is a form element
+    const isFormField = (element) => {
+      if (!(element instanceof HTMLElement)) {
+        return false;
+      }
+
+      const name = element.nodeName.toLowerCase();
+      const type = (element.getAttribute("type") || "").toLowerCase();
+      return (
+        name === "select" ||
+        name === "textarea" ||
+        (name === "input" && ["submit", "reset", "checkbox", "radio"].indexOf(type) < 0) ||
+        element.isContentEditable
+      );
+    };
+
+    const keyEvent = (e) => {
+      if (e.defaultPrevented) return;
+      // Get special keys
+      const keys = `${e.ctrlKey || e.metaKey ? "ctrl+" : ""}${e.altKey ? "alt+" : ""}${e.shiftKey ? "shift+" : ""}`;
+      // If no special keys are pressed and focus is inside a field return
+      if (e.target instanceof Node && isFormField(e.target) && !keys) return;
+      const shortcut = shortcuts[`${keys}${e.code}`];
+      if (shortcut) shortcut(e);
+    };
+
+    window.addEventListener("keydown", keyEvent);
```

Comment:
This may not be required now, so don't [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it), but we may want to add keyboard events of any type, e.g. `keyup`, `keypress` etc.

Reply:
I think it would be easy enough to add an `eventType = "keydown"` down the road without it breaking anything.